relay-event-normalization/src/event.rs

```diff
@@ -753,80 +754,117 @@ fn normalize_performance_score(
-                    // Skip optional components if they are not present on the event.
-                    if component.optional
-                        && !measurements.contains_key(component.measurement.as_str())
-                    {
+            let Some(measurements) = event.measurements.value_mut() else {
+                return;
+            };
+            normalize_performance_score_inner(measurements, profile);
+        }
+    }
+}
+
+/// Computes performance score measurements for a span.
+///
+/// This computes score from vital measurements, using config options to define how it is
+/// calculated.
+pub fn normalize_span_performance_score(
+    browser_name: String,
+    measurements: &mut Measurements,
+    performance_score: Option<&PerformanceScoreConfig>,
+) {
+    let Some(performance_score) = performance_score else {
+        return;
+    };
+    for profile in &performance_score.profiles {
+        if let Some(condition) = &profile.condition {
+            // HACK: we support only checking browser name for spans
+            match condition {
+                RuleCondition::Eq(condition) => {
+                    if condition.name != "event.contexts.browser.name" {
                         continue;
                     }
-                    weight_total += component.weight;
-                }
-                if weight_total.abs() < f64::EPSILON {
-                    // All components are optional or have a weight of `0`. We cannot compute
-                    // component weights, so we bail.
-                    break;
-                }
-                for component in &profile.score_components {
-                    // Optional measurements that are not present are given a weight of 0.
-                    let mut normalized_component_weight = 0.0;
-                    if let Some(value) = measurements.get_value(component.measurement.as_str()) {
-                        normalized_component_weight = component.weight / weight_total;
-                        let cdf = utils::calculate_cdf_score(
-                            value.max(0.0), // Webvitals can't be negative, but we need to clamp in case of bad data.
-                            component.p10,
-                            component.p50,
-                        );
-                        let component_score = cdf * normalized_component_weight;
-                        score_total += component_score;
-                        should_add_total = true;
-
-                        measurements.insert(
-                            format!("score.{}", component.measurement),
-                            Measurement {
-                                value: component_score.into(),
-                                unit: (MetricUnit::Fraction(FractionUnit::Ratio)).into(),
-                            }
-                            .into(),
-                        );
+                    if condition.value != browser_name {
+                        continue;
```

Comment:
If we're gonna hack anyway, I would wrap the span in a pseudo event that has `event.contexts.browser.name` and `measurements` set. That should make the rest of the refactor unnecessary.

Long-term, I would add a new key to the performance score config (e.g. define a field `span_conditions` on `PerformanceScoreProfile` in addition to `conditions` that formulates the condition on a span properly.